packages/eas-cli/src/commands/release/create.ts

```diff
@@ -0,0 +1,97 @@
+import { Command, flags } from '@oclif/command';
+import chalk from 'chalk';
+import gql from 'graphql-tag';
+
+import { graphqlClient, withErrorHandlingAsync } from '../../graphql/client';
+import log from '../../log';
+import { findProjectRootAsync } from '../../project/projectUtils';
+import { promptAsync } from '../../prompts';
+import { getProjectIdAsync } from '../../submissions/commons';
+
+type UpdateRelease = {
+  id: string;
+  releaseName: string;
+};
+
+async function createUpdateReleaseOnAppAsync({
+  appId,
+  releaseName,
+}: {
+  appId: string;
+  releaseName: string;
+}): Promise<UpdateRelease> {
+  const data = await withErrorHandlingAsync(
+    graphqlClient
+      .mutation<
+        { updateRelease: { createUpdateReleaseForApp: UpdateRelease } },
+        { appId: string; releaseName: string }
+      >(
+        gql`
+          mutation CreateUpdateReleaseForApp($appId: ID!, $releaseName: String!) {
+            updateRelease {
+              createUpdateReleaseForApp(appId: $appId, releaseName: $releaseName) {
+                id
+                releaseName
+              }
+            }
+          }
+        `,
+        {
+          appId,
+          releaseName,
+        }
+      )
+      .toPromise()
+  );
+  return data.updateRelease.createUpdateReleaseForApp;
+}
+
+export default class ReleaseCreate extends Command {
+  static description = 'Create a release on the current project.';
+
+  static args = [
+    {
+      name: 'releaseName',
+      required: false,
+      description: 'Name of the release to create',
+    },
+  ];
+
+  async run() {
+    let {
+      args: { releaseName },
+    } = this.parse(ReleaseCreate);
+
+    const projectDir = await findProjectRootAsync(process.cwd());
+    if (!projectDir) {
+      log.error("Please run this command inside your project's directory.");
+      return;
+    }
+    const projectId = await getProjectIdAsync(projectDir);
+
+    if (!releaseName) {
+      const userInput = await promptAsync({
+        type: 'text',
+        name: 'releaseName',
+        message: 'Please name the release:',
+      });
+      releaseName = userInput.releaseName;
+    }
+    if (!releaseName) {
+      log.error('You must specify a releaseName.');
+      return;
+    }
```

Comment:
If I leave the release name empty, it annoyingly exits the whole command:
<img width="842" alt="Screen Shot 2020-11-18 at 15 27 19" src="https://user-images.githubusercontent.com/497214/99536111-91aec680-29b2-11eb-99f3-0afc0d9b54fc.png">


Maybe try this instead? (It will validate that release name is not empty.)
```suggestion
    if (!releaseName) {
      ({ releaseName } = await promptAsync({
        type: 'text',
        name: 'releaseName',
        message: 'Please name the release:',
        validate: value => value ? true : 'Release name may not be empty.',
      }));
    }
```
Perhaps there are other validations we should do too? Max length? Allowed characters?

Also maybe it should show an example of a "good" release name to aid learning?

Reply:
>Perhaps there are other validations we should do too? Max length? Allowed characters?

I was thinking to do this on the server to avoid having multiple different verifications.

> Also maybe it should show an example of a "good" release name to aid learning?

That's a good idea... but what is a good release name? 😅 
One idea we were playing around with was suggesting the git-branch name. But this requires an extra module and also doesn't always work.